chemprop/cli/train.py

```diff
@@ -547,8 +591,8 @@ def main(args):
 
     predictor_ffn = Factory.build(
         predictor_cls,
-        input_dim=mp_block.output_dim + train_dset.d_xf,
-        n_tasks=train_dset.Y.shape[1],
+        input_dim=mp_block.output_dim + train_dset.d_xf if n_components == 1 else mp_block.output_dim + sum(train_dset.datasets[i].d_xf for i in range(n_components)),
+        n_tasks=train_dset.Y.shape[1] if n_components == 1 else train_dset.datasets[0].Y.shape[1],
```

Comment:
```suggestion
        input_dim=mp_block.output_dim + d_xf),
        n_tasks=train_dset.Y.shape[1] if n_components == 1 else train_dset.datasets[0].Y.shape[1],
```
where `d_xf` is defined above:
```
d_df = train_dset.d_xf if n_components == 1 else mp_block.output_dim + sum(dset.d_xf for dset in train_dset.datasets)